helix-core/src/test/java/org/apache/helix/integration/task/TestForceDeleteWorkflow.java

```diff
@@ -0,0 +1,337 @@
+    admin.enableCluster(CLUSTER_NAME, false);
+    Thread.sleep(Disable_Delay);
+
+    _driver.delete(workflowName, true);
+    Thread.sleep(DELETE_DELAY);
+
+    // Enable the cluster before Force Delete
+    admin.enableCluster(CLUSTER_NAME, true);
+
+    // Check that WorkflowConfig, WorkflowContext, and IdealState are indeed deleted for this
+    // workflow
+    Assert.assertNull(_driver.getWorkflowConfig(workflowName));
+    Assert.assertNull(_driver.getWorkflowContext(workflowName));
+    Assert.assertNull(admin.getResourceIdealState(CLUSTER_NAME, workflowName));
+  }
+
+  @Test
+  public void testDeleteRunningWorkflowForcefully() throws Exception {
+    // Create a simple workflow and wait until it reaches the Running state. Then delete the
+    // IdealState,
+    // WorkflowContext and WorkflowConfig.
+    String workflowName = TestHelper.getTestMethodName();
+    Workflow.Builder builder = new Workflow.Builder(workflowName);
+
+    //Workflow Schematic:
+    //               JOB0
+    //                /\
+    //               /  \
+    //             JOB1 JOB2
+
+    JobConfig.Builder jobBuilder0 = JobConfig.Builder.fromMap(WorkflowGenerator.DEFAULT_JOB_CONFIG)
+        .setMaxAttemptsPerTask(1).setWorkflow(workflowName)
+        .setJobCommandConfigMap(ImmutableMap.of(MockTask.JOB_DELAY, "100000"));
+
+    JobConfig.Builder jobBuilder1 = JobConfig.Builder.fromMap(WorkflowGenerator.DEFAULT_JOB_CONFIG)
+        .setMaxAttemptsPerTask(1).setWorkflow(workflowName)
+        .setJobCommandConfigMap(ImmutableMap.of(MockTask.JOB_DELAY, "100000"));
+
+    JobConfig.Builder jobBuilder2 = JobConfig.Builder.fromMap(WorkflowGenerator.DEFAULT_JOB_CONFIG)
+        .setMaxAttemptsPerTask(1).setWorkflow(workflowName)
+        .setJobCommandConfigMap(ImmutableMap.of(MockTask.JOB_DELAY, "100000"));
+
+    builder.addParentChildDependency("JOB0", "JOB1");
+    builder.addParentChildDependency("JOB0", "JOB2");
+    builder.addJob("JOB0", jobBuilder0);
+    builder.addJob("JOB1", jobBuilder1);
+    builder.addJob("JOB2", jobBuilder2);
+
+    _driver.start(builder.build());
+
+    // Wait until workflow is created and goes to running stage.
+    boolean isWorkflowRunning = TestHelper.verify(() -> {
+      WorkflowContext wCtx1 = _driver.getWorkflowContext(workflowName);
+      return (wCtx1 != null && wCtx1.getWorkflowState() == TaskState.IN_PROGRESS);
+    }, 60 * 1000);
+    Assert.assertTrue(isWorkflowRunning);
+
+    // Check that WorkflowConfig, WorkflowContext, and IdealState are indeed created for this
+    // workflow
+    Assert.assertNotNull(_driver.getWorkflowConfig(workflowName));
+    Assert.assertNotNull(_driver.getWorkflowContext(workflowName));
+    Assert.assertNotNull(admin.getResourceIdealState(CLUSTER_NAME, workflowName));
+
+    // Disable the cluster before Force Delete
+    admin.enableCluster(CLUSTER_NAME, false);
+    Thread.sleep(Disable_Delay);
+
+    _driver.delete(workflowName, true);
```

Comment:
How do you ensure that the workflow is in progress at the time of deletion?

Reply:
The execution time for each job is set for a large time. So I think it is safe to consider that running at the time of deletion.

Reply:
Could you check that the tasks are in RUNNING state then?